### ISSUE TITLE
Extend SA1129 to cover parameterless struct constructors

### ISSUE THREAD

**Issue by Arthri** [DotNetAnalyzers/StyleCopAnalyzers#3854]
C#10 introduced the ability to declare and use [parameterless struct constructors](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/parameterless-struct-constructors), which means `new Struct()` and `default(Struct)` can be different. Consider the following code sample,
```cs
using System;

X.L(); // 12
X.K(); // 0

internal struct X
{
    public int I;

    public X()
    {
        I = 12;
    }

    public static void L()
    {
        Console.WriteLine(new X().I);
    }
    
    public static void K()
    {
        Console.WriteLine(default(X).I);
    }
}
```

I believe that SA1129 "A value type was constructed using the syntax `new T()`." should not be removed, rather, it should check whether the struct defines a parameterless constructor before raising any warnings

**Comment by bjornhellander** [comment]
Duplicate of #3430
